.github/scripts/depexts/generate-actions.sh

```diff
@@ -83,35 +83,41 @@ EOF
     ;;
   opensuse)
   # glpk-dev is installed manually because os-family doesn't handle tumbleweed
-    cat >$dir/Dockerfile << EOF
+    cat > "$dir/Dockerfile" << EOF
 FROM opensuse/leap:15.3
 RUN zypper --non-interactive install $mainlibs $ocaml diffutils gzip glpk-devel
 RUN zypper --non-interactive install gcc-c++
 EOF
     ;;
   oraclelinux)
-    cat >$dir/Dockerfile << EOF
+    cat > "$dir/Dockerfile" << EOF
 FROM oraclelinux:8
 RUN yum install -y $mainlibs
 RUN yum install -y gcc-c++
 EOF
   ;;
   ubuntu)
-  cat >$dir/Dockerfile << EOF
+  cat > "$dir/Dockerfile" << EOF
 FROM ubuntu:20.04
 RUN apt update
 RUN apt install -y $mainlibs $ocaml
 RUN apt install -y g++
 EOF
     ;;
+  nix)
+    cat > "$dir/Dockerfile" << EOF
+FROM nixos/nix
+RUN nix-channel --update
+RUN nix-env -iA nixpkgs.gnum4 nixpkgs.git nixpkgs.rsync nixpkgs.patch nixpkgs.bzip2 nixpkgs.gnumake nixpkgs.wget nixpkgs.ocaml nixpkgs.ocamlPackages.ocaml-compiler-libs nixpkgs.unzip nixpkgs.gcc nixpkgs.diffutils nixpkgs.patch nixpkgs.getconf nixpkgs.gnused nixpkgs.gawk
```

Comment:
haha `nixpkgs.ocamlPackages.ocaml-compiler-libs` is in fact JaneStreet's `compiler-libs` wrapper library, not the compiler's compiler-libs as a separate package (which is otherwise frequently split as a separate package in some distributions for some reason)
```suggestion
RUN nix-env -iA nixpkgs.gnum4 nixpkgs.git nixpkgs.rsync nixpkgs.patch nixpkgs.bzip2 nixpkgs.gnumake nixpkgs.wget nixpkgs.ocaml nixpkgs.unzip nixpkgs.gcc nixpkgs.diffutils nixpkgs.patch nixpkgs.getconf nixpkgs.gnused nixpkgs.gawk
```